### PR TITLE
ADC improvement & fix

### PR DESCRIPTION
**Pull request by rds76**
## Description:

- bugfix
- improvement: 
   - option to add additional reading for better accuracy
   - calibration constant
   - option to publish at least at defined interval

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

File: main/ZsensorADC.ino [hunk 1]
```diff
@@ -49,14 +49,26 @@ void MeasureADC() {
     timeadc = millis();
     static int persistedadc;
     int val = analogRead(ADC_GPIO);
+    int sum_val = val;
+    if (NumberOfReadingsADC > 1) { // add extra measurement for accuracy
+      for (int i=1; i<NumberOfReadingsADC; i++){
+        delay(7);
```
Comment (1technophile) [inline main/ZsensorADC.ino]: Is there a way to average the accuracy without having a blocking `delay()` ? I'm afraid about the impacts on the main loop and on other modules.

Comment (rds76) [reply main/ZsensorADC.ino]: Well, its just 7ms per read so for recommended 10 reads its 70ms in total (still optional) and guess `delay` actually yields every ms. Btw check RFB, theres is delay in loop on sending as well
Currently I'm running that (10 readings) with RF and DS1820 temp sensor with no issues so far...
Maybe we can short the delay to 1ms or remove it at all and also add one dummy analogRead as its recommended for 8266:

> the reliability of the readings can be significantly improved by discarding an initial reading and then averaging the results 

I'm afraid average values across loops adds another noise for already poor ADC in 8266 so better to read all in once

**Comment by rds76** [comment]
updated PR, readings are much more consistent without delay in adcread loop (prolly caused by yield() in delay() creating noise)
_with 1ms delay_
```
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":368,"adc_reads":10,"adc_sum":3680,"mvolt_scaled":1140}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":368,"adc_reads":10,"adc_sum":3681,"mvolt_scaled":1141}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":368,"adc_reads":10,"adc_sum":3684,"mvolt_scaled":1142}
house/OpenMQTTGateway/ADCtoMQTT {"adc":368,"adc_reads":10,"adc_sum":3683,"mvolt_scaled":1141}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
```
_no delay_
```
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
house/OpenMQTTGateway/ADCtoMQTT {"adc":363,"adc_reads":10,"adc_sum":3630,"mvolt_scaled":1125}
```

**Comment by 1technophile** [comment]
Thanks!

**Pull request merged**